_docs/backend/openapi.yaml

```
@@ -0,0 +1,332 @@
+openapi: 3.0.3
+info:
+  title: HTGDF
+  description: API description between the frontend (e.g. WebApp) and the backend providing the game logic.
```

```suggestion
  description: API for managing the logic of a HTGDF game.
```

The speicification should not care how its used

---

Since our system has three well-defined components (frontend/gui, backend/logic, backend/datamodel) the API spec may as well specify at which point between this components it is supposed to be used. Since we do not plan on exposing the API publicly (e.g. with CORS) this description is sufficient.

---

Yes but all APIs are used by any kind of client so it does not provide any benefit of mentioning it :)

---

I will keep it anyway, since I believe the documentation should also reference in which scenario it can be applied. For most openapi-Specs this is any client (and therefore they do not mention anything), but for our case the client is ALWAYS responsible for the GUI (even if it is another implementation), so we should keep the general reference, that the client is an UI.